lighthouse-core/config/config.js

```diff
@@ -491,70 +501,56 @@ class Config {
   }
 
   /**
-   * Filter out any unrequested items from the config, based on requested top-level categories.
-   * @param {LH.Config.Json} oldConfig Lighthouse config object
-   * @param {!Array<string>=} categoryIds ID values of categories to include
-   * @param {!Array<string>=} auditIds ID values of categories to include
-   * @param {!Array<string>=} skipAuditIds ID values of categories to exclude
-   * @return {LH.Config.Json} A new config
+   * Filter out any unrequested items from the config, based on requested categories or audits.
+   * @param {Config} config
    */
-  static generateNewFilteredConfig(oldConfig, categoryIds, auditIds, skipAuditIds) {
-    // 0. Clone config to avoid mutating it
-    const config = deepCloneConfigJson(oldConfig);
-    config.audits = Config.expandAuditShorthandAndMergeOptions(config.audits);
-    config.passes = Config.expandGathererShorthandAndMergeOptions(config.passes);
-    config.passes = Config.requireGatherers(config.passes);
+  static filterConfigIfNeeded(config) {
+    const settings = config.settings;
+    if (!settings.onlyCategories && !settings.onlyAudits && !settings.skipAudits) {
+      return config;
```

Comment:
want to add a return typedef? seems like we don't actually want to return anything

Reply:
> seems like we don't actually want to return anything

ah, yes, good call